jetty-http2/http2-server/src/main/java/org/eclipse/jetty/http2/server/AbstractHTTP2ServerConnectionFactory.java

```diff
@@ -245,18 +252,43 @@ protected ServerParser newServerParser(Connector connector, ServerParser.Listene
         return new ServerParser(connector.getByteBufferPool(), listener, getMaxDynamicTableSize(), getHttpConfiguration().getRequestHeaderSize());
     }
 
-    private class ConnectionListener implements Connection.Listener
+    private class HTTP2SessionContainer implements Connection.Listener, Dumpable
     {
+        private final Set<Session> sessions = ConcurrentHashMap.newKeySet();
+
         @Override
         public void onOpened(Connection connection)
         {
-            addManaged((LifeCycle)((HTTP2Connection)connection).getSession());
+            Session session = ((HTTP2Connection)connection).getSession();
+            sessions.add(session);
+            LifeCycle.start(session);
```

Comment:
if start throws an exception, will onClosed be called to remove it?

Reply:
Historically, if an exception was thrown from a `Connection.Listener` we let it unwind the stack, which I don't think it's ideal.
I think such listeners should be wrapped in `try/catch`, log the exception and continue.
So either we wrap, or we leave it as is. I'm for wrapping.